pkg/cli/repositories.go

```diff
@@ -0,0 +1,447 @@
+		usr, err := os.UserHomeDir()
+		if err != nil {
+			SetError(err, 1)
+			return
+		}
+
+		pathCheck := filepath.Join(usr, ".vorteil", "repository-keys")
+
+		if _, err = os.Stat(pathCheck); os.IsNotExist(err) {
+			fmt.Printf("no keys found\n")
+			return
+		}
+
+		// Open default
+		f, err := os.Open(filepath.Join(pathCheck, "default"))
+		if err != nil {
+			SetError(err, 2)
+			return
+		}
+		defer f.Close()
+
+		h := md5.New()
+		if _, err := io.Copy(h, f); err != nil {
+			SetError(err, 3)
+			return
+		}
+
+		fis, err := ioutil.ReadDir(pathCheck)
+		if err != nil {
+			SetError(err, 4)
+			return
+		}
+
+		for _, fi := range fis {
+
+			if fi.Name() != "default" {
+				f2, err := os.Open(filepath.Join(pathCheck, fi.Name()))
+				if err != nil {
+					SetError(err, 5)
+					return
+				}
+				h2 := md5.New()
+				if _, err := io.Copy(h2, f2); err != nil {
+					SetError(err, 6)
+					return
+				}
+				if bytes.Equal(h.Sum(nil), h2.Sum(nil)) {
+					fmt.Println(fi.Name())
+				}
+				f2.Close()
+			}
+
+		}
+	},
+}
+
+var createKeyCmd = &cobra.Command{
+	Use:   "create NAME TOKEN",
+	Short: "Create keys saves a file with the token to be referenced using a name",
+	Args:  cobra.MaximumNArgs(2),
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) < 2 {
+			return errors.New("must provide a NAME and TOKEN")
+		}
+		usr, err := os.UserHomeDir()
+		if err != nil {
+			return err
+		}
+		// Check if key storage exists if not create it
+		pathCheck := filepath.Join(usr, ".vorteil", "repository-keys")
+		_, err = os.Stat(pathCheck)
+		if err != nil {
+			err = os.MkdirAll(pathCheck, os.ModePerm)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		name := args[0]
+		key := args[1]
+
+		usr, err := os.UserHomeDir()
+		if err != nil {
+			SetError(err, 1)
+			return
+		}
+		pathCheck := filepath.Join(usr, ".vorteil", "repository-keys")
+
+		// Check if file exists
+		if !flagForce {
+			// if stat returns no error return error saying you need to provide the force flag
+			_, err := os.Stat(filepath.Join(pathCheck, name))
+			if err == nil {
+				SetError(errors.New("key file already exists provide --force to overwrite"), 2)
+				return
+			}
+		}
+
+		// Write key to a file under that keys directory
+		err = ioutil.WriteFile(filepath.Join(pathCheck, name), []byte(key), os.ModePerm)
+		if err != nil {
+			SetError(err, 3)
+			return
+		}
+
+		// Check if default and write another file called default under repository-keys
+		if flagDefault {
+			err = ioutil.WriteFile(filepath.Join(pathCheck, "default"), []byte(key), os.ModePerm)
+			if err != nil {
+				SetError(err, 4)
+				return
+			}
+		}
+	},
+}
+
+func init() {
+	f := createKeyCmd.Flags()
+	f.BoolVar(&flagDefault, "default", false, "save this key to use as default")
+	f.BoolVar(&flagForce, "force", false, "force overwrite of key file")
+}
+
+var listKeysCmd = &cobra.Command{
```

Comment:
Not needed, but it would be cool if you can see which key is the default in the list command.

Reply:
Something like:
```sh
$ ./borteil repositories keys list
key1
key3
vrepo [default]
```